src/routes/cart.routes.ts

```diff
@@ -15,6 +22,12 @@ cartRouter.get("/", [protectRoute], viewCart);
 
 cartRouter.delete("/remove/:id", [protectRoute], removeFromCart);
 
-cartRouter.delete("/clear", [protectRoute], clearCart);
+cartRouter.delete("/clear", [isAuth], clearCart);
+
+cartRouter.patch(
+  "/update/:id",
+  [protectRoute, ValidateJoi(ProductSchema.product.updateCart)],
```

Comment:
This middleware only works for `sellers`.
Why are you using it for buyers as well?
I think you should use `isAuth` instead.